coap-gateway/service/refreshToken.go

```diff
@@ -71,24 +71,24 @@ func validUntilToExpiresIn(validUntil time.Time) int64 {
 }
 
 // the device doesn't provide an authorization provider during refresh token so we need to do try refresh against all providers in parallel
-func refreshAccessToken(ctx context.Context, server *Service, refreshToken string) (*oauth2.Token, error) {
+func refreshAccessToken(ctx context.Context, client *Client, refreshToken string) (oauth2.Token, error) {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
 	var wg sync.WaitGroup
 	var mutex sync.Mutex
 	var token *oauth2.Token
 	var err error
-	for name, p := range server.providers {
+	for name, p := range client.server.providers {
 		wg.Add(1)
 		provider := p
-		errSubmit := server.taskQueue.Submit(func() {
+		errSubmit := client.server.taskQueue.Submit(func() {
 			defer wg.Done()
-			tokenTmp, errTmp := provider.Refresh(ctx, refreshToken)
+			tokenTmp, errTmp := client.tokenCache.Refresh(ctx, provider, refreshToken)
```

Comment:
This can be called from multiple threads so `client.tokenCache.Refresh` must be thread-safe.

Reply:
The `refreshTokenPostHandler` is not serialized like the `signUpPostHandler`?